qiime2/plugin/__init__.py

```diff
@@ -23,4 +23,308 @@
            'Float', 'Metadata', 'MetadataColumn', 'Categorical', 'Numeric',
            'Properties', 'Range', 'Start', 'End', 'Choices', 'Visualization',
            'Jobs', 'Threads', 'TypeMap', 'TypeMatch', 'ValidationError',
-           'Citations', 'CitationRecord', 'get_available_cores']
+           'Citations', 'CitationRecord', 'get_available_cores',
+           'SingleFileDirectoryFormat']
+
+
+Set = Set
+"""**Deprecated** - use List or Collection instead
+
+A set of unique elements without a defined order except when used with
+a semantic type, in which case, the views are provided to the plugin as a list.
+"""
+
+List = List
+"""An alias for a :py:data:`.Collection` with numeric auto-incrementing keys.
+
+Examples
+--------
+>>> from qiime2.plugin import List, Str
+
+A regular list:
+
+>>> ['a', 'b', 'c'] in List[Str]
+True
+
+A collection is also compatible:
+
+>>> {'arbitrary_key': 'a'} in List[Str]
+True
+"""
+
+Collection = Collection
+"""An ordered set of key-value pairs. Compatible with lists and dictionaries.
+
+The keys of a collection are always strings, and the values are defined by the
+variant provided to the type-field.
+
+Whenever a list is provided, it will be coerced into a dictionary with
+auto-incrementing integer keys.
+
+Examples
+--------
+>>> from qiime2.plugin import Collection, Str
+
+A regular dictionary:
+
+>>> {'key1': 'a', 'key2': 'b'} in Collection[Str]
+True
+
+A list:
+
+>>> ['a', 'b'] in Collection[Str]
+True
+"""
+
+
+Visualization = Visualization
+"""The type of a QIIME 2 Visualization.
+
+This is not a semantic type as it represents a terminal/non-composable output.
+
+An output with this type provides no assurances about the structure of the data
+as it is meant for human interpretation.
+
+Examples
+--------
+>>> from qiime2.plugin import Visualization
+>>> Visualization
+Visualization
+"""
+
+Int = Int
+"""
+An integer without any particular bounds.
+It can use the predicates :py:class:`.Range`, :py:func:`.Start`, and
```

Comment:
I think of it more as "using" a predicate, since the predicate is literally an attribute attached to the whole type expression. i.e. 
```python
>>> type = Int % Range(0, 5)
>>> type.predicate
Range(0, 5)
```